corehq/apps/style/static/style/ko/components/inline_edit.js

```diff
@@ -0,0 +1,143 @@
+ *  - rows: Number of rows in textarea.
+ *
+ * By default, the widget is client-side only, and it is up to the calling code to actually save the value
+ * (likely by providing the widget with a name or id). The following parameters may be used to implement
+ * a widget capable of saving to the server via ajax.
+ *  - url: The URL to call on save.
+ *  - saveValueName: Name to associate with text value when saving. Defaults to 'value'.
+ *  - saveParams: Any additional data to pass along. May contain observables.
+ *  - errorMessage: Message to display if server returns an error.
+ */
+
+hqDefine('style/ko/components/inline_edit.js', function() {
+    return {
+        viewModel: function(params) {
+            var self = this;
+
+            // Attributes passed on to the input
+            self.name = params.name || '';
+            self.id = params.id || '';
+
+            // Data
+            self.placeholder = params.placeholder || '';
+            self.original = (ko.isObservable(params.value) ? params.value() : params.value) || '';
+            self.value = ko.isObservable(params.value) ? params.value : ko.observable(self.original);
+            self.lang = params.lang || '';
+
+            // Styling
+            self.inline = params.inline || false;
+            self.rows = params.rows || 2;
+
+            // Interaction: determine whether widget is in read or write mode
+            self.editing = ko.observable(false);
+            self.saveHasFocus = ko.observable(false);
+            self.cancelHasFocus = ko.observable(false);
+
+            // Save to server
+            self.url = params.url;
+            self.errorMessage = params.errorMessage || gettext("Error saving, please try again");
+            self.saveParams = ko.utils.unwrapObservable(params.saveParams) || {};
+            self.saveValueName = params.saveValueName || 'value';
+            self.hasError = ko.observable(false);
+            self.isSaving = ko.observable(false);
+
+            // On edit, set editing mode, which controls visibility of inner components
+            self.edit = function() {
+                self.editing(true);
+            };
+
+            self.save = function() {
+                // Client save: just store the value and switch modes
+                self.original = self.value();
+                self.editing(false);
+
+                // Server save
+                if (self.url) {
+                    var data = self.saveParams;
+                    _.each(data, function(value, key) {
+                        data[key] = ko.utils.unwrapObservable(value);
+                    });
+                    data[self.saveValueName] = self.value();
+                    self.isSaving(true);
+                    $.ajax({
+                        url: self.url,
+                        type: 'POST',
+                        dataType: 'JSON',
+                        data: data,
+                        success: function (data) {
```

Comment:
It is nice to know, but what I mean is that it's fairly common practice to accept arguments that you're not going to use, particularly for callbacks passed to a library like jquery.  Just a bit weird/annoying to flag that as a potential error.

Looks like it's quite configurable though: http://eslint.org/docs/rules/no-unused-vars#options